RocketChat/Rocket.Chat.ReactNative
# fix: Link preview should not enlarge small images

## Proposed changes

This pull request addresses an issue with image previews being resized disproportionately in the message content. The changes ensure that images are displayed in their original dimensions, preserving their aspect ratios and preventing unnecessary enlargement or shrinkage.

The solution includes:
- Fetching the image's original dimensions using `Image.getSize`.
- Dynamically setting the width and height of the image based on its original dimensions.
- Managing the loading state of the image to handle errors and display the image correctly once loaded.

## Issue(s)

Fixes #5509 

## How to test or reproduce

1. Create a message containing a link to an image, particularly a very small one.
2. Observe the image preview in the message content.

**Expected Result:** 
The image should display in its original size without being disproportionately enlarged.

## Screenshots

Before:
![Screenshort_1234](https://private-user-images.githubusercontent.com/404840/299107339-186c055d-975d-4eab-9d9e-ce22a845f10b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA4ODgwNTcsIm5iZiI6MTcyMDg4Nzc1NywicGF0aCI6Ii80MDQ4NDAvMjk5MTA3MzM5LTE4NmMwNTVkLTk3NWQtNGVhYi05ZDllLWNlMjJhODQ1ZjEwYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxM1QxNjIyMzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNmRlNGNkYzc5ZmE3NDliNjA1NmQ4ODZlOTVjMWE5YjNhMDFhZDM2ZDVmNWFhYTMzMWI1OWQ0NTc4ODRhM2ZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NK64KkSHFHrrLuWEkla5KUlsmzBChsJTDAgU-vWU9Wg)

After:

![Screenshot_1720886956](https://github.com/user-attachments/assets/894cf9c7-463e-4798-a27f-a5ff3e4789a0)

## Types of changes
